### PR TITLE
Removed "proposals" folder

### PR DESCRIPTION
**Pull request by buskamuza**
1. Removed "proposals" folder, as `proposal` == `PR`, but the result is a design document, so everything should go to the Design Documents folder eventually.
   1. Fixes https://github.com/magento/architecture/issues/22
1. Added process for PRs that need update after feedback.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,25 +26,23 @@ facilitator
...
 1. Create a PR with the new or updated document to discuss
 1. Share the PR with internal team(s) and the Magento community through existing channels (Twitter, Slack, blog post, etc)
    1. Suggestion: include deadline for receiving feedback
 1. Get a feedback. We expect a feedback as:
    1. Comments into the PR thread
    1. Likes/dislikes/other emojis to the comments or the PR itself. We kindly encourage contributors to submit explanations about pros and cons that they have noticed.
 1. Update the PR to the received feedback accordingly or submit a reply if the proposed changes are not applicable with clear explanation.
-1. When all participants of the discussion have come to agreement and confirmed their approval, add "approved" label to the PR.
-1. A Magento core architect merges the PR. 
+   1. Add `needs update` label while the PR is in work, so it's not
```
Comment (paliarush) [inline README.md]: `so it's not` what?

File: README.md [hunk 2]
```diff
@@ -26,25 +26,23 @@ facilitator
 
 ## The Workflow
 
-1. Fork the repository and add or edit a document in your branch
-   1. Design Documents must be stored in `design-documents` directory. Contributions are expected from Magento core engineers mostly, although the community members are able to contribute.
-   1. Proposals are contributed into the `proposals` directory. Contributions are expected from both Magento core engineers and the community members.
-   1. We may add more categories in future
+1. Fork the repository and add or edit a document in your branch.
+Contributions are expected from Magento core engineers mostly, although the community members are able to contribute.
```
Comment (melnikovi) [inline README.md]: Maybe change to `are able to contribute as well` or `can and encouraged to contribute as well`?

File: README.md (hunk 1 above)
Comment (buskamuza) [reply README.md]: Good question :) Removed unnecessary phrase.

File: README.md (hunk 2 above)
Comment (buskamuza) [reply README.md]: Updated

**Pull request merged**